app/src/main/java/nl/tudelft/cs4160/trustchain_android/main/OverviewConnectionsActivity.java

```diff
@@ -329,25 +420,22 @@ public void run() {
     private void startListenThread() {
         final Context context = this;
 
-        Thread listenThread = new Thread(new Runnable() {
-            @Override
-            public void run() {
-                try {
-                    ByteBuffer inputBuffer = ByteBuffer.allocate(BUFFER_SIZE);
-                    while (!Thread.interrupted()) {
-                        inputBuffer.clear();
-                        SocketAddress address = network.receive(inputBuffer);
-                        inputBuffer.flip();
-                        network.dataReceived(context, inputBuffer, (InetSocketAddress) address);
-                    }
-                } catch (IOException e) {
-                    e.printStackTrace();
-                    Log.d("App-To-App Log", "Listen thread stopped");
+        Thread listenThread = new Thread(() -> {
+            try {
+                ByteBuffer inputBuffer = ByteBuffer.allocate(BUFFER_SIZE);
+                while (!Thread.interrupted()) {
```

Comment:
Send and receive threads never exit. This is also the case in the current develop branch but I think this PR is a good place to add this, otherwise we can make a new issue.
The channel is closed in OnDestroy but both threads keep running and throwing ClosedChannelExceptions

Reply:
on second thought lets just fix this in a new PR